### ISSUE TITLE
FrequencySketch is useful but not public

### ISSUE THREAD

**Issue by acruise** [ben-manes/caffeine#223]
In my current project I think I really want to use FrequencySketch, to maintain a saturating,  automatically aging population count of distinct elements in an infinite stream, but at the moment I don't have any real need to build a cache per se.  Can you see any major downside to flipping it to public?

**Comment by ben-manes** [comment]
Yes, I would like to reduce the visibility of implementation details and only expose the API. This way I can modify algorithms and optimize it for my needs, rather than provide a generalized interface. You are welcome to copy it into your own project.

When I started this project, I thought it might expand beyond caching so I made more things public (e.g. `UnsafeAccess`, `SingleConsumerQueue`, and a few now hidden). That was a mistake as its complex as is and the singular focus is perhaps better, so I'd like to hide those internal leaks in the next major version.

One justification is that we've been experimenting with adaptive algorithms (I can send a conference paper privately if interested). So that might be built into the sketch to automatically tune the window/main sizes based on an indicator. The ability to rework internals without forking or worrying about breaking someone else is nice.
